### PR TITLE
feat: close with state reason

### LINKED ISSUE #299: Close issue as not_planned
**Is your feature request related to a problem? Please describe.**
Right now, the `$close()` built-in, by default, closes the issue as "completed" which is not correct for when we close issues for being stale.

**Describe the solution you'd like**
Add a string arg to the `$close()` built-in signature that is the closure state of the issue which can only be "completed" or "not_planned".

Currently we are using the GitHub v45. We need to update to v46 in order to proceed with this solution - https://github.com/google/go-github/releases/tag/v46.0.0.

**Describe alternatives you've considered**
*None*

**Additional context**
*None*

### PR DESCRIPTION
**Pull request by zolamk**
## Description
This PR introduces an option second parameter to the `$close` built-in action, which allows us to specify the reason an issue was closed.
- add second parameter to `$close` built-in action

## Related issue

Closes #299 

## Type of change

New feature (non-breaking change which adds functionality)

## How was this tested?
Manual & Unit Tests.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

**Comment by marcelosousa** [comment]
Hi @zolamk could you please separate the dependency upgrade from the functionality in this PR? Thanks!

**Comment by zolamk** [comment]
> Hi @zolamk could you please separate the dependency upgrade from the functionality in this PR? Thanks!

Sure @marcelosousa

File: engine/transform.go [hunk 1]
```diff
@@ -69,7 +69,12 @@ func addDefaultPullRequestCountBy(str string) string {
 }
 
 func addEmptyCloseComment(str string) string {
-	return strings.ReplaceAll(str, "$close()", "$close(\"\")")
+	return strings.ReplaceAll(str, "$close()", "$close(\"\", \"completed\")")
+}
+
+func addCompletedCloseReason(str string) string {
+	r := regexp.MustCompile(`\$close\("([^"]*)"\)`)
+	return r.ReplaceAllString(str, `$$close("$1", "completed")`)
```
Comment (shay2025) [inline engine/transform.go:77]: Isn't there a duplicate `$`?

**Review by shay2025 — CHANGES_REQUESTED** [review]
Some minor suggestions.

File: engine/transform.go (hunk 1 above)
Comment (zolamk) [reply engine/transform.go:77]: the `ReplaceAllString` method treats `$` as a capture group reference so you have to put 2 of them if you want a literal `$` in there

File: lang/aladino/mocks.go [hunk 2]
```diff
@@ -343,14 +379,36 @@ func MockDefaultEnv(
...
+	githubClient := MockDefaultGithubClient(ghApiClientOptions, ghGraphQLHandler)
+
+	mockedEnv, err := mockEnvWith(prOwner, prRepoName, prNum, githubClient, eventPayload, builtIns, targetEntity)
+	if err != nil {
+		t.Fatalf("[MockDefaultEnvWithTargetEntity] failed to create mock env: %v", err)
+	}
+
+	return mockedEnv
+}
+
```
[the remark this answers is not in the document]
Comment (zolamk) [reply lang/aladino/mocks.go:411]: Yeah, touching the `MockDefaultEnv` method required changes in a lot of places, but I can do it now or later if you prefer

[the remark this answers is not in the document]
Comment (shay2025) [reply lang/aladino/mocks.go:411]: Let's do it in a separate PR :slightly_smiling_face:

**Review by shay2025 — APPROVED** [review]

**Pull request merged**